core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -81,6 +81,8 @@ class BrokerServer(
   val controllerQuorumVotersFuture: CompletableFuture[util.Map[Integer, AddressSpec]]
 ) extends KafkaBroker {
 
+  val lifecycleManager: BrokerLifecycleManager = new BrokerLifecycleManager(config, time, threadNamePrefix)
```

Comment:
This change will make some tests fail since `lifecycleManager` is stateful(for example, initialCatchUpFuture ) but we are stopping and restarting it in `LeaderElectionCommandTest`.

Reply:
thanks for the heads up. I will go with a different fix.